ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -280,6 +293,49 @@ - (void)testShouldHandleVnodeOpEvent {
     XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
 }
 
+- (void)testCurrentProcessWasSpawnedByRegularUser {
+    // Defaults should pass for all tests
+    XCTAssertTrue(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Process tree with all UIDs less than 500 were not spawned by regular users
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddProcess(1 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+    XCTAssertFalse(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Test a process with a service UID, valid parent pid, but proc_find fails to find parent pid
+    MockCalls::Clear();
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddProcess(1 /*pid*/, 1 /*credentialId*/, 501 /*ppid*/, "test" /*name*/);
+    XCTAssertFalse(CurrentProcessWasSpawnedByRegularUser());
+    XCTAssertTrue(MockCalls::DidCallFunction(KextMessageLogged, KEXTLOG_ERROR));
+    MockProcess_Reset();
+
+    // 'sudo' scenario: Root process with non-root parent
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddCredential(2 /*credentialId*/, 501 /*UID*/);    
+    MockProcess_AddProcess(1 /*pid*/, 1 /*credentialId*/, 501 /*ppid*/, "test" /*name*/);
+    MockProcess_AddProcess(501 /*pid*/, 2 /*credentialId*/, 502 /*ppid*/, "test" /*name*/);
+    XCTAssertTrue(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Process and it's parent are service users
```

Comment:
Nit:  In this comment you use the term "service users" and in the above comment you use the term "non-root parent".  Could you make these consistently one way or the other ("service" is the term used the code so maybe use that here too)?